feature.koa/bsande6/Rm8eK7r-tkOu4CjdeWrJumQAPvl4/OAuth

bsande6 opened the pull request:
Creating an event subscription has not been properly tested due to the inability to dynamically create the publication url for localhost. Additionally the cache table is named "qualtrics_backends_api_tokens" and will need to be created manually.

Need to manually run `./manage.py lms createcachetable --settings=devstack_docker` to create the cache table in the database from within the LMS Docker container using the `make lms-shell` terminal command.

Commit:
Updated html display of grading problems

Commit:
Added is_answered field to keep track of survey status

Commit:
Started working on OAuth

Commit:
Added functionalty for OAuth2

ztraboo commented on qualtricssurvey/qualtrics_api.py:
```diff
@@ -92,15 +114,42 @@ def get_survey_response(self, survey_id, response_id):
+        if token_cached is not None:
+            return token_cached
+        else:
+            url = settings.QUALTRICS_OAUTH_URL
```
@bsande6 Recommend changing this to the following to avoid having to set this additional URL in settings.
```
url = "{}/auth2/token/{}".format(self._api_surveys_base_url)
```

ztraboo replied:
Updated the code with these recommended changes.

ztraboo reviewed (changes requested):
Minor changes needed in the QualtricsApi class for connecting with the API.

ztraboo committed:
Updated the QualtricsAPI to have correct endpoint for OAuth2 authentication.

- Added in separate helper method `_api_auth_url()` for OAuth endpoint and used that in the `get_oauth_token` method.
- Updated helper method for `_api_base_url()` to include `/API/` path
- Removed database model `SurveyStatus` to account for XBlock `is_answered` field taking its place.
